python/Ganga/new_tests/GPI/GangaUnitTest.py

```diff
@@ -162,7 +179,7 @@ def setUp(self, gangadir=None, wipe_repo=None, extra_opts=[]):
         # This is called before each unittest
         if gangadir is None:
             import os
-            gangadir = os.path.join('$HOME/gangadir_testing', self.__class__.__name__)
+            gangadir = os.path.join('$HOME/gangadir_testing')  # TODO make separate dirs per test suite, `self.__class__.__name__`
```

Comment:
At the moment the test harness thinks each test is being run with its own gangadir and so it's removing those directories as it goes. However, each test actually always runs in the gangadir of the first test that was run (at least for the registry) and so things are not being cleaned up correctly. This is because the registry location-setting start-up code is only run once and it is never updated. If we fix that in the future then we can go back to running with a gangadir per test.